### PR TITLE
Changes and fixes

### PR DESCRIPTION
**Pull request by MisterDA**
Hi! This is a set of patches that switches to the latest Dune version, and improves the C stub with Unicode wide-string and macros for manipulating option types, that were introduced in more-or-less recent versions of OCaml.

### CODE REVIEW

**Comment by avsm** [comment]
@MisterDA could we trouble you to also add a ocaml/setup-ocaml Github action so that this can be tested on Windows? Or if @djs55 could test that would also permit a merge (I don't have a Windows box handy atm)

**Comment by MisterDA** [comment]
Added a few more stuff:
- use ocaml-setup for CI (remove travis and appveyor);
- wrap the library (with a transition message);
- documentation fixes;
- lower bound on OCaml 4.06;
- use ocamlformat.

The CI passes successfully on my fork: https://github.com/MisterDA/ocaml-win-eventlog/actions.

**Comment by MisterDA** [comment]
Hey @avsm, I've added setup-ocaml support and some more fixes. Could you consider merging and releasing it?
The CI passes on my fork: https://github.com/MisterDA/ocaml-win-eventlog/actions/runs/1181730200.

**Comment by avsm** [comment]
Thanks! @djs55 do you have time to merge/release this?

File: dune-project [hunk 1]
```diff
@@ -1,2 +1,29 @@
-(lang dune 1.0)
+(lang dune 2.9)
 (name win-eventlog)
+(use_standard_c_and_cxx_flags true)
+(generate_opam_files true)
+
+(source (github mirage/ocaml-win-eventlog))
+(license ISC)
+(authors "David Scott")
```
Comment (djs55) [inline dune-project:8]: Feel free to add yourself to `(authors ...)`

**Review by djs55 — APPROVED** [review]
Thanks, looks good to me!

**Comment by djs55** [comment]
Changes are in https://github.com/ocaml/opam-repository/pull/19376

**Pull request merged**